packages/util/test/equal.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { equal, equalArrays, equalObjects } from '../src/equal';
+describe('equal', () => {
+  describe('equal', () => {
+    it('returns true for same values', () => {
+      expect(equal(1, 1)).toBe(true);
+      expect(equal(undefined, undefined)).toBe(true);
+      expect(equal(null, null)).toBe(true);
+      expect(equal('', '')).toBe(true);
+    });
+
+    it('returns false for different values', () => {
+      expect(equal(1, 2)).toBe(false);
+      expect(equal(undefined, null)).toBe(false);
+      expect(equal(null, '')).toBe(false);
+      expect(equal('1', 1 as any)).toBe(false);
+    });
+
+    it('returns true for equal nested arrays', () => {
+      expect(equal([[1, 2]], [[1, 2]])).toBe(true);
+    });
+    it('returns true for equal nested objects', () => {
+      expect(equal([[1, { a: 'a' }]], [[1, { a: 'a' }]])).toBe(true);
+    });
+
+    it('returns false for nullish values', () => {
+      expect(equal([0, 1, '2'], ['', 1, '2'])).toBe(false);
+    });
+
+    it('returns false for differing nested arrays and object', () => {
```

Comment:
I removed one of the tests because it was redundant. The other tests are about nested arrays and objects, which is functionality implemented in `equal` but I guess in the end it is simple enough that it doesn't really matter.